commit 66dca9b8c50b5e59d3bea8b21cee5c6dae6c9c46
Author: Joe Malicki <jmalicki@metacarta.com>
Date:   Thu Aug 14 17:14:48 2008 -0400

    [SCSI] megaraid_sas: remove sysfs dbg_lvl world writeable permissions
    
    /sys/bus/pci/drivers/megaraid_sas/dbg_lvl defaults to being
    world-writable, which seems bad (letting any user affect kernel driver
    behavior and logging level).
    
    This turns off group and user write permissions, so that on typical
    production systems only root can write to it.
    
    [jejb: fix up rejections]
    Signed-off-by: Joseph Malicki <jmalicki@metacarta.com>
    Acked-by: "Yang, Bo" <Bo.Yang@lsi.com>
    Signed-off-by: James Bottomley <James.Bottomley@HansenPartnership.com>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.24.orig/drivers/scsi/megaraid/megaraid_sas.c linux-source-2.6.24/drivers/scsi/megaraid/megaraid_sas.c
--- linux-source-2.6.24.orig/drivers/scsi/megaraid/megaraid_sas.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/drivers/scsi/megaraid/megaraid_sas.c	2010-01-31 19:51:22.000000000 -0700
@@ -3018,7 +3018,7 @@ megasas_sysfs_set_dbg_lvl(struct device_
 	return retval;
 }
 
-static DRIVER_ATTR(dbg_lvl, S_IRUGO|S_IWUGO, megasas_sysfs_show_dbg_lvl,
+static DRIVER_ATTR(dbg_lvl, S_IRUGO|S_IWUSR, megasas_sysfs_show_dbg_lvl,
 		   megasas_sysfs_set_dbg_lvl);
 
 /**
